public/app/plugins/datasource/loki/components/LokiContextUi.tsx

```diff
@@ -74,6 +75,13 @@ export function LokiContextUi(props: LokiContextUiProps) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [contextFilters, initialized]);
 
+  useEffect(() => {
+    return () => {
+      clearTimeout(timerHandle.current);
```

Comment:
Could we now remove this from the useEffect above? 
```
    return () => {
      clearTimeout(timerHandle.current);
    };
```

Reply:
Hm, not really. We are clearing it above to start a new timer, to "stage" changes. Here we are clearing it because it's not needed anymore as the whole component is unmounted.